include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -48,358 +94,883 @@ struct Basic
 namespace surface_interactions
 {
 
-template<class RayDirInfo>
-struct Isotropic
+#define NBL_CONCEPT_NAME Isotropic
+#define NBL_CONCEPT_TPLT_PRM_KINDS (typename)(typename)(typename)
+#define NBL_CONCEPT_TPLT_PRM_NAMES (T)(B)(U)                        // B is type Basic<T>
+#define NBL_CONCEPT_PARAM_0 (iso,T)
+#define NBL_CONCEPT_PARAM_1 (normV,B)
+#define NBL_CONCEPT_PARAM_2 (normN,vector<U, 3>)
```

Comment:
I haven't checked, but `(normV,typename T::ray_dir_info_type)` and `(normN,typename T::vector_type)` should work